### PR TITLE
[IMP] mail: remove top level sudo in _message_format

### PR DESCRIPTION
**Pull request by seb-odoo**
Some specific fields need sudo(), not all of them. Only keep necessary sudo with justifications.

Part of task-3605717

### CODE REVIEW

File: addons/mail/models/mail_message.py [hunk 1]
```diff
@@ -963,13 +963,25 @@ def _message_format(self, format_reply=True, msg_vals=None, for_current_user=Fal
                 scheduled_dt_by_msg_id[scheduler.mail_message_id.id] = scheduler.scheduled_datetime
         record_by_message = self._record_by_message()
         for vals in vals_list:
-            message_sudo = self.browse(vals['id']).sudo().with_prefetch(self.ids)
+            message = self.browse(vals["id"]).sudo().with_prefetch(self.ids)
```
Comment (ThanhDodeurOdoo) [inline addons/mail/models/mail_message.py:966]: the sudo is still there

**Comment by seb-odoo** [comment]
Apparently this can only be done after the fix in https://github.com/odoo/odoo/pull/171869

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+

**Review by seb-odoo — COMMENTED** [review]
@robodoo r+

**Pull request closed without merging**